### PR TITLE
chore: helper param and input builder cleanup

### PR DESCRIPTION
**Pull request by kibo-susantasamanta**

### CODE REVIEW

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Review by kibo-amolpawar — COMMENTED** [review]
Could you please rename test files with params?
"buildAddOrRemoveWishlistItem" to "buildAddOrRemoveWishlistItemParams"

File: lib/types/PaymentTypes.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Card, Contact } from '../gql/types'
+import { Contact } from '../gql/types'
```
Comment (TriptiVermaXP) [inline lib/types/PaymentTypes.ts:1]: add type

File: lib/helpers/buildCreateWishlistItemInputParams.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import getConfig from 'next/config'
 
 import type { WishlistInput } from '@/lib/gql/types'
-export const buildCreateWishlistItemInput = (
+export const buildCreateWishlistItemInputParams = (
```
Comment (TriptiVermaXP) [inline lib/helpers/buildCreateWishlistItemInputParams.ts:4]: Didn't find use of buildCreateWishlistItemInputParams anywhere, please check.

**Review by TriptiVermaXP — APPROVED** [review]
Please check minor comments

File: hooks/queries/useProducts/useProducts.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { useQuery } from 'react-query'
 
 import { makeGraphQLClient } from '@/lib/gql/client'
 import { searchProductsQuery } from '@/lib/gql/queries'
-import { buildProductSearchInput } from '@/lib/helpers/buildProductSearchInput'
+import { buildProductSearchInputParams } from '@/lib/helpers/buildProductSearchInputParams'
```
Comment (kibo-sushant) [inline hooks/queries/useProducts/useProducts.ts:5]: Not using index file?

File: lib/api/operations/get-product-search.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import { fetcher } from '@/lib/api/util'
 import { searchProductsQuery } from '@/lib/gql/queries'
-import { buildProductSearchInput } from '@/lib/helpers/buildProductSearchInput'
+import { buildProductSearchInputParams } from '@/lib/helpers/buildProductSearchInputParams'
```
Comment (kibo-sushant) [inline lib/api/operations/get-product-search.ts:3]: Not using index file for import?

File: lib/helpers/buildAddOrRemoveWishlistItemParams.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { productGetters } from '../getters'
 
 import type { ProductCustom } from '../types'
 
-export const buildAddOrRemoveWishlistItem = (product: ProductCustom) => {
+export const buildAddOrRemoveWishlistItemParams = (product: ProductCustom) => {
```
Comment (kibo-sushant) [inline lib/helpers/buildAddOrRemoveWishlistItemParams.ts:5]: return type missing

File: lib/helpers/buildOrdersFilterInputParams.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ const getPastDateTimestamp = (months: number) => {
   return today.toISOString()
 }
 
-export const buildOrdersFilterInput = (params: {
+export const buildOrdersFilterInputParams = (params: {
```
Comment (kibo-sushant) [inline lib/helpers/buildOrdersFilterInputParams.ts:11]: return type missing

Comment (sss-bct) [reply lib/helpers/buildOrdersFilterInputParams.ts:11]: Done

File: lib/types/PaymentTypes.ts (hunk 1 above)
Comment (sss-bct) [reply lib/types/PaymentTypes.ts]: Done

File: lib/helpers/buildCreateWishlistItemInputParams.ts (hunk 2 above)
Comment (sss-bct) [reply lib/helpers/buildCreateWishlistItemInputParams.ts]: It's there now again updated as per Sushant's suggestion to buildCreateWishlistItemparams.

File: components/checkout/ShippingStep/ShippingStep.tsx [hunk 7]
```diff
@@ -88,13 +87,9 @@ const ShippingStep = (props: ShippingProps) => {
...
   const handleAddressValidationAndSave = () => setValidateForm(true)
 
-  const updateShippingInfo = async (params: ShippingParams) =>
-    await updateCheckoutShippingInfo.mutateAsync(params)
-
   const handleSaveAddress = async ({ contact }: { contact: Contact }) => {
-    const params = buildCheckoutShippingParams({ checkout, contact: contact })
     try {
-      await updateShippingInfo(params)
+      await updateCheckoutShippingInfo.mutateAsync({ checkout, contact: contact })
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/checkout/ShippingStep/ShippingStep.tsx]: Done.

File: components/checkout/PaymentStep/PaymentStep.tsx [hunk 8]
```diff
@@ -28,7 +28,7 @@ import {
 import { PaymentType, PaymentWorkflow } from '@/lib/constants'
 import { addressGetters, cardGetters, checkoutGetters, accountDetailsGetters } from '@/lib/getters'
 import { tokenizeCreditCardPayment } from '@/lib/helpers'
-import { buildCardPaymentActionForCheckoutInput } from '@/lib/helpers/buildPaymentActionForCheckoutInput'
+import { buildCardPaymentActionForCheckoutInputParams } from '@/lib/helpers/buildCardPaymentActionForCheckoutInputParams'
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/checkout/PaymentStep/PaymentStep.tsx]: Done.

File: hooks/mutations/useUpdateCheckout/useUpdateCheckoutShippingInfo/useUpdateCheckoutShippingInfo.ts [hunk 9]
```diff
@@ -2,6 +2,10 @@ import { useMutation, useQueryClient } from 'react-query'
 
 import { makeGraphQLClient } from '@/lib/gql/client'
 import { setShippingInfo } from '@/lib/gql/mutations'
+import {
+  buildCheckoutShippingParams,
+  CheckoutShippingParams,
+} from '@/lib/helpers/buildCheckoutShippingParams'
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply hooks/mutations/useUpdateCheckout/useUpdateCheckoutShippingInfo/useUpdateCheckoutShippingInfo.ts:8]: Done

File: hooks/mutations/useUpdateCheckout/useUpdateCheckoutShippingInfo/useUpdateCheckoutShippingInfo.ts [hunk 10]
```diff
@@ -11,9 +15,11 @@ export interface ShippingInfo {
   fulfillmentInfoInput: FulfillmentInfoInput
 }
 
-const updateShippingInfo = async (shippingInfo: ShippingInfo) => {
+const updateShippingInfo = async (params: CheckoutShippingParams) => {
   const client = makeGraphQLClient()
 
+  const shippingInfo: ShippingInfo = buildCheckoutShippingParams(params)
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply hooks/mutations/useUpdateCheckout/useUpdateCheckoutShippingInfo/useUpdateCheckoutShippingInfo.ts]: Removed.

File: hooks/queries/useProducts/useProducts.ts (hunk 3 above)
Comment (sss-bct) [reply hooks/queries/useProducts/useProducts.ts]: Done

File: lib/api/operations/get-product-search.ts (hunk 4 above)
Comment (sss-bct) [reply lib/api/operations/get-product-search.ts]: Done

File: lib/helpers/buildAddOrRemoveWishlistItemParams.ts (hunk 5 above)
Comment (sss-bct) [reply lib/helpers/buildAddOrRemoveWishlistItemParams.ts]: Done

File: components/page-templates/ProductListingTemplate/ProductListingTemplate.tsx [hunk 11]
```diff
@@ -207,7 +207,7 @@ const ProductListingTemplate = (props: ProductListingTemplateProps) => {
 
   const handleWishList = async (product: ProductCustom) => {
     try {
-      const addOrRemoveWishlistItemParams = buildAddOrRemoveWishlistItem(product)
+      const addOrRemoveWishlistItemParams = buildAddOrRemoveWishlistItemParams(product)
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/page-templates/ProductListingTemplate/ProductListingTemplate.tsx]: Removed.

**Review by kibo-amolpawar — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**